components/style/properties/longhand/inherited_box.mako.rs

```diff
@@ -50,6 +50,131 @@ ${helpers.single_keyword("image-rendering",
+            if let Some(angle) = self.angle {
+                try!(angle.to_css(dest));
+                if self.flipped {
+                    dest.write_str(" flipped")
+                } else {
+                    Ok(())
+                }
+            } else {
+                if self.flipped {
+                    dest.write_str("flipped")
+                } else {
+                    dest.write_str("from-image")
+                }
+            }
+        }
+    }
+
+    pub mod computed_value {
+        use values::specified::Angle;
+
+        #[derive(Clone, PartialEq, Copy, Debug)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub enum T {
+            FromImage,
+            AngleWithFlipped(Angle, bool),
+        }
+    }
+
+    const INITIAL_ANGLE: Angle = Angle(0f32);
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T::AngleWithFlipped(INITIAL_ANGLE, false)
+    }
+
+    use values::CSSFloat;
+    use std::f32::consts::PI;
+    const TWO_PI: CSSFloat = 2f32*PI;
+
+    #[inline]
+    fn normalize_angle(angle: &Angle) -> Angle {
+        let radians = angle.radians();
+        let normalized_radians = ((radians % TWO_PI) + TWO_PI) % TWO_PI;
+        Angle::from_radians(normalized_radians)
+    }
+
+    impl ToComputedValue for SpecifiedValue {
+        type ComputedValue = computed_value::T;
+
+        #[inline]
+        fn to_computed_value(&self, _: &Context) -> computed_value::T {
+            if let Some(ref angle) = self.angle {
+                let normalized_angle = normalize_angle(angle);
+                computed_value::T::AngleWithFlipped(normalized_angle, self.flipped)
+            } else {
+                if self.flipped {
+                    computed_value::T::AngleWithFlipped(INITIAL_ANGLE, true)
+                } else {
+                    computed_value::T::FromImage
+                }
+            }
+        }
+
+        #[inline]
+        fn from_computed_value(computed: &computed_value::T) -> Self {
+            match *computed {
+                computed_value::T::FromImage => SpecifiedValue { angle: None, flipped: false },
+                computed_value::T::AngleWithFlipped(angle, flipped) =>
+                    SpecifiedValue { angle: Some(angle), flipped: flipped },
+            }
+        }
+    }
+
+    impl ToCss for computed_value::T {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            match *self {
+                computed_value::T::FromImage => dest.write_str("from-image"),
+                computed_value::T::AngleWithFlipped(angle, flipped) => {
+                    try!(angle.to_css(dest));
+                    if flipped {
+                        try!(dest.write_str(" flipped"));
+                    }
+                    Ok(())
+                },
+            }
+        }
+    }
+
+    pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
```

Comment:
I used the parameter names from the [property hacking guide](https://github.com/servo/servo/wiki/Property-hacking-guide). Will change.